### DISCUSSION TITLE
this is amazing.but im having issues when trying to install

### DISCUSSION THREAD

**Discussion by djwugee** [JeffreyCA/spleeter-web#818] [Q&A]
hey man any further details on how to install this, redis is running, but django and python keeps saying i need to be in virtualenv, but when in virtualenv modules arent being found even after  pip install.
maybe im doing it all wrong. but any help would be good,, thanks👍

**Comment by JeffreyCA** [comment]
Hi, are you following the [Docker instructions](https://github.com/JeffreyCA/spleeter-web#getting-started-with-docker)?

**Comment by djwugee** [comment]
hey man, yea i did but .sorry man im new to docker.and just spent the past few days trying to figire it out. so ive followed the instructions and. right now i cant remember the exact errors..but i know its something to do with the ports. when i try to visit 127.0.0.1:80 i get a page with just some txt saying forbidden
anyway this reply is just to say thanks for responding.
when i get back to it ill get more details.and if you could help me out.thatd be awsome man

**Reply by JeffreyCA** [reply]
Can you post the contents of your Terminal/Command Prompt? The text or screenshot is fine

**Reply by djwugee** [reply]
hey man..so i got it running, but i cant access the frontend @ 0.0.0.0:8000, 127.... or 192....
just get error: site cant be reached

heres a small snippet of my terminal

spleeter-web-api-1          | 
spleeter-web-api-1          | 10 static files copied to '/webapp/staticfiles', 160 unmodified, 761 post-processed.
spleeter-web-api-1          | Applying migrations
spleeter-web-api-1          | 2023-10-26 03:41:04,094 [nnabla][INFO]: Initializing CPU extension...
spleeter-web-api-1          | /usr/local/lib/python3.8/site-packages/norbert/contrib.py:3: DeprecationWarning: Please use `gaussian_filter1d` from the `scipy.ndimage` namespace, the `scipy.ndimage.filters` namespace is deprecated.
spleeter-web-api-1          |   from scipy.ndimage.filters import gaussian_filter1d
spleeter-web-api-1          | No changes detected in app 'api'
spleeter-web-api-1          | 2023-10-26 03:41:39,767 [nnabla][INFO]: Initializing CPU extension...
spleeter-web-api-1          | /usr/local/lib/python3.8/site-packages/norbert/contrib.py:3: DeprecationWarning: Please use `gaussian_filter1d` from the `scipy.ndimage` namespace, the `scipy.ndimage.filters` namespace is deprecated.
spleeter-web-api-1          |   from scipy.ndimage.filters import gaussian_filter1d
spleeter-web-api-1          | Operations to perform:
spleeter-web-api-1          |   Apply all migrations: admin, api, auth, contenttypes, sessions
spleeter-web-api-1          | Running migrations:
spleeter-web-api-1          |   No migrations to apply.
spleeter-web-api-1          | Starting server
spleeter-web-api-1          | [2023-10-26 03:41:53 +0000] [102] [INFO] Starting gunicorn 20.1.0
spleeter-web-api-1          | [2023-10-26 03:41:53 +0000] [102] [INFO] Listening at: http://0.0.0.0:8000 (102)
spleeter-web-api-1          | [2023-10-26 03:41:53 +0000] [102] [INFO] Using worker: sync
spleeter-web-api-1          | [2023-10-26 03:41:53 +0000] [104] [INFO] Booting worker with pid: 104

**Reply by JeffreyCA** [reply]
It should be running at 127.0.0.1:80. Can you share the full `docker compose` command you're running?

**Reply by djwugee** [reply]
so ive had to change the port its running on due to another server running. but i must be doing something wrong with the port setup.
 docker-compose -f docker-compose.yml -f docker-compose.build.yml -f docker-compose.prod.yml -f docker-compose.prod.selfhost.yml up --build

and heres my docker ps
e8b62d10ed32   jeffreyca/spleeter-web-nginx:latest     "/docker-entrypoint.…"   5 minutes ago   Up 4 minutes   0.0.0.0:443->443/tcp, 80/tcp, 0.0.0.0:3000->3000/tcp  

 spleeter-web-nginx-1
5737ca5d317a   jeffreyca/spleeter-web-backend:latest   "api-entrypoint.sh"      5 minutes ago   Up 4 minutes   6000/tcp                                               

spleeter-web-api-1
69b11331a886   jeffreyca/spleeter-web-backend:latest   "./celery-slow-entry…"   5 minutes ago   Up 4 minutes                                                          

spleeter-web-celery-slow-1
79479d134c35   jeffreyca/spleeter-web-backend:latest   "./celery-fast-entry…"   5 minutes ago   Up 4 minutes                                                          

spleeter-web-celery-fast-1
c4b5fbdbe8a9   redis:6.0-buster                        "docker-entrypoint.s…"   3 hours ago     Up 5 minutes   6379/tcp                                               spleeter-web-redis-1

**Reply by JeffreyCA** [reply]
What's your `.env` configuration? I think you'll have to set `NGINX_PORT` to your desired port.

E.g. if you set `NGINX_PORT=81` then it should be accessible at `127.0.0.1:81` instead of port 80.

**Reply by djwugee** [reply]
my  .env only has
NGINX_PORT=81
YOUTUBE_API_KEY=ytdatav3apikey

which now tells me its running on port 81
but still no access to frontend on 127.0.0.1:81.

would my fix be anything to do with these?
ALLOW_ALL_HOSTS
APP_HOST
API_HOST

**Reply by JeffreyCA** [reply]
You can try setting `ALLOW_ALL_HOSTS=1`, but it should be running on port 81 with your current .env configuration so I'm not sure why you're unable to access the frontend.

A few other troubleshooting questions:

- What OS are you running Docker on?
- Does 0.0.0.0:81 or localhost:81 work?

Any errors in the docker output when you try to access 127.0.0.1:81? Normally you should see something like:
![image](https://github.com/JeffreyCA/spleeter-web/assets/9157833/311a48f2-7582-4701-a85c-26ebe8a32463)

**Reply by djwugee** [reply]
im running docker on mac
127...:81 192... 0... or localhost not working and heres a screenshot of my docker
it looks like theres no requests being receieved
thanks for your help
![Screen Shot 2023-10-28 at 6 17 54 PM](https://github.com/JeffreyCA/spleeter-web/assets/34990515/46f3b620-f0ff-4816-b6a8-d92ce1d0913e)

**Reply by JeffreyCA** [reply]
Can you share the `docker ps` output again? Also you should see the port binding like so:

![image](https://github.com/JeffreyCA/spleeter-web/assets/9157833/f64b4df6-77d8-4999-8518-4a64a6f8dbce)


One other thing to try is modifying `docker-compose.prod.yml`, replace `${NGINX_PORT:-80}` with the port you want:

```yml
services:
  nginx:
    ports:
      - "82:80"
      - "${NGINX_PORT_SSL:-443}:443"
```

**Reply by djwugee** [reply]
yea ive tried that aswell to no avail.
im gonna try on my windowns machine today and hopefully it works.
thanks for ur help man!👍
